sdk/src/logs/logger.cc

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "opentelemetry/sdk/logs/logger.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace sdk
+{
+namespace logs
+{
+class LoggerProvider;
```

Comment:
This `class LoggerProvider;` declaration is not needed because it is declared in `opentelemetry/sdk/logs/logger.h` which is included by this file.

Reply:
This is used so that the Logger(LoggerProvider) constructor is able to find LoggerProvider before it uses it. Is there another way around this?

Reply:
I tried removing it but it doesn't compile without it. The reason its there is because the logger.h and logger_provider.h both include each other, so we need a forward declaration to tell the compiler that the class exists and will be defined later (or something like that :)).

Reply:
Emm, are removing this line or the same declaration in `logger.h`? I just removed this line and produced build successfully. I meant this forward declaration is unnecessary because it was already forwarded declared [here](https://github.com/open-telemetry/opentelemetry-cpp/pull/386/files#diff-66a32cfcf1db2f0ca92857788c2c37ad1b1ae6c4913e8b560e1e63f939b10b21R35)

Reply:
Thats right, thanks for the catch!